clients/filesystem-fuse/src/fuse_api_handle.rs

```diff
@@ -0,0 +1,507 @@
+use futures_util::stream::BoxStream;
+use futures_util::StreamExt;
+use std::ffi::{OsStr, OsString};
+use std::num::NonZeroU32;
+use std::time::{Duration, SystemTime};
+
+pub(crate) struct FuseApiHandle<T: RawFileSystem> {
+    fs: T,
+    default_ttl: Duration,
+    fs_context: FileSystemContext,
+}
+
+impl<T: RawFileSystem> FuseApiHandle<T> {
+    const DEFAULT_ATTR_TTL: Duration = Duration::from_secs(1);
+    const DEFAULT_MAX_WRITE_SIZE: u32 = 16 * 1024;
+
+    pub fn new(fs: T, context: FileSystemContext) -> Self {
+        Self {
+            fs: fs,
+            default_ttl: Self::DEFAULT_ATTR_TTL,
+            fs_context: context,
+        }
+    }
+
+    pub async fn get_file_path(&self, file_id: u64) -> String {
+        self.fs.get_file_path(file_id).await
+    }
+
+    async fn get_modified_file_stat(
+        &self,
+        file_id: u64,
+        size: Option<u64>,
+        atime: Option<Timestamp>,
+        mtime: Option<Timestamp>,
+    ) -> Result<FileStat, Errno> {
+        let mut file_stat = self.fs.stat(file_id).await?;
+
+        if let Some(size) = size {
+            file_stat.size = size;
+        };
+
+        if let Some(atime) = atime {
+            file_stat.atime = atime;
+        };
+
+        if let Some(mtime) = mtime {
+            file_stat.mtime = mtime;
+        };
+
+        Ok(file_stat)
+    }
+}
+
+impl<T: RawFileSystem> Filesystem for FuseApiHandle<T> {
+    async fn init(&self, _req: Request) -> fuse3::Result<ReplyInit> {
+        self.fs.init().await;
+        Ok(ReplyInit {
+            max_write: NonZeroU32::new(Self::DEFAULT_MAX_WRITE_SIZE).unwrap(),
+        })
+    }
+
+    async fn destroy(&self, _req: Request) {
+        //TODO need to call the destroy method of the local_fs
+    }
+
+    async fn lookup(
+        &self,
+        _req: Request,
+        parent: Inode,
+        name: &OsStr,
+    ) -> fuse3::Result<ReplyEntry> {
+        let name = name.to_string_lossy();
+        let file_stat = self.fs.lookup(parent, &name).await?;
+        Ok(ReplyEntry {
+            ttl: self.default_ttl,
+            attr: fstat_to_file_attr(&file_stat, &self.fs_context),
+            generation: 0,
+        })
+    }
+
+    async fn getattr(
+        &self,
+        _req: Request,
+        inode: Inode,
+        fh: Option<u64>,
+        _flags: u32,
+    ) -> fuse3::Result<ReplyAttr> {
+        // check the opened file file_id is the same as the inode
+        if let Some(fh) = fh {
+            self.fs.valid_file_id(inode, fh).await?;
+        }
+
+        let file_stat = self.fs.stat(inode).await?;
+        Ok(ReplyAttr {
+            ttl: self.default_ttl,
+            attr: fstat_to_file_attr(&file_stat, &self.fs_context),
+        })
+    }
+
+    async fn setattr(
+        &self,
+        _req: Request,
+        inode: Inode,
+        fh: Option<u64>,
+        set_attr: SetAttr,
+    ) -> fuse3::Result<ReplyAttr> {
+        // check the opened file file_id is the same as the inode
```

Comment:
please add the comment to the interface of `valid_file_id `, no need to add the comment for the invoked places.
change the comment to `Verify the equality between the file_id of the opened file and the inode.`？